### PR TITLE
Update handler scheduler according to the original

### PR DESCRIPTION
**Pull request by kornelijepetak**
## What's this?
Update HandlerScheduler according to the original source.

## Why do we want this?
If they considered it as a bug, we should as well.

## How is it done?
By following suit.

📝 **EDIT.** This PR also changes:
* all instances of `SynchronizationContext.Current` replaced by DI-based scheduler.
* `GetAwaiter().GetResult()` changed into `SubscribeOn(main)` to avoid deadlocks when crossing sync/async boundary.

### Side effects
All actions (regardless of the thread) are now posted to the handler thread.

## :squid: Permissions
Guys who were present on a CALI meeting.

### CODE REVIEW

File: Toggl.Droid/Startup/HandlerScheduler.cs [hunk 1]
```diff
@@ -2,31 +2,32 @@
...
+    /// See https://bit.ly/2JVR63C
+    /// </remarks>
     public sealed class HandlerScheduler : IScheduler
     {
-        private readonly long looperId;
         private readonly Handler handler;
 
         public DateTimeOffset Now => DateTimeOffset.Now;
 
         public HandlerScheduler(Handler handler, long? threadIdAssociatedWithHandler)
```
Comment (daividssilverio) [inline Toggl.Droid/Startup/HandlerScheduler.cs]: Might as well remove the `threadIdAssociatedWithHandler`, right?

File: Toggl.Droid/Startup/HandlerScheduler.cs [hunk 2]
```diff
@@ -2,31 +2,32 @@
...
-using Toggl.Core.Analytics;
 
 namespace Toggl.Droid
 {
+    /// <summary>
+    /// Custom handler scheduler
+    /// </summary>
+    /// <remarks>
+    /// Roughly based on reactiveui/ReactiveUI HandlerSchedulerClass
+    /// See https://bit.ly/2JVR63C
```
Comment (juanlaube) [inline Toggl.Droid/Startup/HandlerScheduler.cs]: I'd prefer the full link instead of bit.ly, you never know when this could stop working.

**Comment by daividssilverio** [comment]
🤔 I've tracked the newly introduced freezes to usages of `FirstAsync().GetAwaiter().GetResult();` like in the `Login` and `SignUp` activities; which can be replaced by `FirstAsync().Subscribe(observer).DisposedBy(disposeBag)`.

I don't know why the changes have caused the freezes though. If we are to merge this, the other usages of `FirstAsync().GetAwaiter().GetResult()` also have to be fixed.

File: Toggl.Core.UI/ViewModels/Settings/SettingsViewModel.cs [hunk 3]
```diff
@@ -482,10 +482,10 @@ private async Task selectBeginningOfWeek()
             syncManager.InitiatePushSync();
         }
 
-        private async Task checkCalendarPermissions()
+        private void checkCalendarPermissions()
         {
-            var authorized = permissionsChecker.CalendarPermissionGranted.FirstAsync().GetAwaiter().GetResult();
-            calendarPermissionGranted.OnNext(authorized);
+            permissionsChecker.CalendarPermissionGranted.FirstAsync()
+                .Subscribe(calendarPermissionGranted.OnNext);
```
Comment (daividssilverio) [inline Toggl.Core.UI/ViewModels/Settings/SettingsViewModel.cs]: Please add a `.DisposedBy(disposeBag`).

File: Toggl.Droid/Activities/LoginActivity.cs [hunk 4]
```diff
@@ -33,8 +33,13 @@ protected override void OnCreate(Bundle bundle)
 
             InitializeViews();
 
-            emailEditText.Text = ViewModel.Email.FirstAsync().GetAwaiter().GetResult();
-            passwordEditText.Text = ViewModel.Password.FirstAsync().GetAwaiter().GetResult();
+            ViewModel.Email.FirstAsync()
+                .SubscribeOn(AndroidDependencyContainer.Instance.SchedulerProvider.MainScheduler)
+                .Subscribe(emailEditText.Rx().TextObserver());
```
Comment (daividssilverio) [inline Toggl.Droid/Activities/LoginActivity.cs]: Same here.

File: Toggl.Droid/Activities/LoginActivity.cs [hunk 5]
```diff
@@ -33,8 +33,13 @@ protected override void OnCreate(Bundle bundle)
...
+
+            ViewModel.Password.FirstAsync()
+                .SubscribeOn(AndroidDependencyContainer.Instance.SchedulerProvider.MainScheduler)
+                .Subscribe(passwordEditText.Rx().TextObserver());
```
Comment (daividssilverio) [inline Toggl.Droid/Activities/LoginActivity.cs]: And here.

File: Toggl.Droid/Activities/SignUpActivity.cs [hunk 6]
```diff
@@ -33,8 +33,13 @@ protected override void OnCreate(Bundle bundle)
 
             InitializeViews();
 
-            emailEditText.Text = ViewModel.Email.FirstAsync().GetAwaiter().GetResult();
-            passwordEditText.Text = ViewModel.Password.FirstAsync().GetAwaiter().GetResult();
+            ViewModel.Email.FirstAsync()
+                .SubscribeOn(AndroidDependencyContainer.Instance.SchedulerProvider.MainScheduler)
+                .Subscribe(emailEditText.Rx().TextObserver());
```
Comment (daividssilverio) [inline Toggl.Droid/Activities/SignUpActivity.cs]: And so on.

**Pull request merged**